src/conversions.c

```diff
@@ -280,12 +280,19 @@ to_unicode(PyArray_Descr *descr,
  * Convert functions helper for the generic converter.
  */
 static PyObject *
-call_converter_function(PyObject *func, const Py_UCS4 *str, size_t length)
+call_converter_function(
+        PyObject *func, const Py_UCS4 *str, size_t length, bool byte_converters)
 {
     PyObject *s = PyUnicode_FromKindAndData(PyUnicode_4BYTE_KIND, str, length);
     if (s == NULL || func == NULL) {
         return s;
     }
+    if (byte_converters) {
+        Py_SETREF(s, PyUnicode_AsEncodedString(s, "latin1", NULL));
+        if (s == NULL) {
+            return NULL;
+        }
+    }
```

Comment:
Will have to move this up for `dtype="S"` support if I move the implementation for string length discovery to python.